cmd/sops/common/common.go

```diff
@@ -0,0 +1,84 @@
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"io/ioutil"
+
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/codes"
+	"go.mozilla.org/sops/keyservice"
+	"gopkg.in/urfave/cli.v1"
+)
+
+type DecryptTreeOpts struct {
+	Tree        *sops.Tree
+	Stash       map[string][]interface{}
+	KeyServices []keyservice.KeyServiceClient
+	IgnoreMac   bool
+	Cipher      sops.DataKeyCipher
+}
+
+func DecryptTree(opts DecryptTreeOpts) ([]byte, error) {
+	dataKey, err := opts.Tree.Metadata.GetDataKeyWithKeyServices(opts.KeyServices)
+	if err != nil {
+		return nil, cli.NewExitError(err.Error(), codes.CouldNotRetrieveKey)
```

Comment:
This package is part of the CLI, it's just splitting up code that was previously in one file in `cmd/sops/main.go`.

Reply:
What I'm trying to say is that `cli` is an implementation detail of the cmd, and the sub-packaged should throw generic errors instead of going through the `cli` package